common/lib/capa/capa/capa_problem.py

```diff
@@ -808,7 +825,13 @@ def _extract_html(self, problemtree):  # private
         # otherwise, render children recursively, and copy over attributes
         tree = etree.Element(problemtree.tag)
         for item in problemtree:
-            item_xhtml = self._extract_html(item)
+            item_xhtml = self._extract_html(item, question_id)
+
+            if item.tag == 'question':
+                item_xhtml.set('class', 'question')
+                item_xhtml.set('id', str(question_id))
```

Comment:
Should we namespace this IDref a little bit to avoid collisions?  q1, q2, q3 maybe?  Also, @muhammad-ammar do you see any reason why this value wouldn't be available to templates that inherit from this one?  IDrefs are going to be critical inside the individual problem templates and having a unique IDref to build related IDrefs from would be very helpful.